servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -224,16 +230,17 @@ def _process_component_relations(self, instance_config, batch_size, timeout, rel
                     data.update({"tags": instance_tags})
 
                     self.relation(parent_sys_id, child_sys_id, relation_type, data)
-            completed = len(state) < batch_size
+            completed = total_relations < batch_size
             offset += batch_size
 
     def _get_json_batch(self, url, offset, batch_size, timeout, auth):
         if "?" not in url:
             limit_args = "?"
         else:
             limit_args = "&"
-        limit_args = limit_args + "sysparm_query=ORDERBYsys_created_on&sysparm_offset={}&sysparm_limit={}".\
-            format(offset, batch_size)
+        limit_args = "{}sysparm_query=ORDERBYsys_created_on&sysparm_offset={}&sysparm_limit={}".format(limit_args,
```

Comment:
Better to not reuse a variable like this. Maybe call the `limit_ags` above, something like `query_delimiter`?

Reply:
fixed